resources/web/wwi/FloatingIde.js

```diff
@@ -0,0 +1,15 @@
+import FloatingWindow from './FloatingWindow.js';
+
+export default class FloatingIDE extends FloatingWindow {
+  constructor(parentNode, name, url) {
+    super(parentNode, name, url);
+
+    this.headerText.innerHTML = 'IDE';
+
+    this.frame.id = this.name + '-robot-window';
+    const elements = url.split('/').filter(element => element);
+    let portString = elements[elements.length - 1];
+    let port = parseFloat(portString) + 500;
```

Comment:
```suggestion
    const portString = elements[elements.length - 1];
    const port = parseFloat(portString) + 500;
```
My bad, I forgot to make them `const`